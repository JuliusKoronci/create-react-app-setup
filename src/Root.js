// @flow
import React from 'react';
import { Switch, Route, BrowserRouter } from 'react-router-dom';
import createHistory from 'history/createBrowserHistory';
import { ConnectedRouter } from 'react-router-redux';
import { Provider } from 'react-redux';
import configureStore from './store/configureStore';
import { Homepage, Dashboard } from './pages';

const history = createHistory();
const store = configureStore();

const Root = () => (
	<div>
		<Provider store={store}>
			<ConnectedRouter history={history}>
				<BrowserRouter>
					<Switch>
						<Route exact path='/' component={ Homepage } />
						<Route exact path='/dashboard' component={ Dashboard } />
					</Switch>
				</BrowserRouter>
			</ConnectedRouter>
		</Provider>
	</div>
);
export default Root;
