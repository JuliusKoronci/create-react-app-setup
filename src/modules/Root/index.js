// @flow
import React from 'react';
import {Provider} from 'react-redux';
import {
    Route,
} from 'react-router-dom';
import {ConnectedRouter} from 'react-router-redux';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import store, {history} from '../../store/configureStore';
import {RouteItems, RouteItem} from '../../FlowTypes';
import routes from '../../routes';

const Root = () => {
    const renderRoutes = (routeList: Array<RouteItems>) => routeList.map((route: RouteItem) => (
        <Route key={route.path} exact path={route.path} component={route.component()}>
            {route.children && renderRoutes(route.children)}
        </Route>
    ));

    return (
        <Provider store={store}>
            <MuiThemeProvider>
                <ConnectedRouter history={history}>
                    <div>
                        {renderRoutes(routes)}
                    </div>
                </ConnectedRouter>
            </MuiThemeProvider>
        </Provider>
    );
};

export default Root;
