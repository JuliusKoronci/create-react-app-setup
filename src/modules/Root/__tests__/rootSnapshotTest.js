import React from 'react';
import Root from '../index';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const wrapper = shallow(
	<Root />
);
describe('Root DemoView', () => {
	it('Should match snapshot', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});

