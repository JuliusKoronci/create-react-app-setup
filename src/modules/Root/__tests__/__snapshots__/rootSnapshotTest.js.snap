// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root DemoView Should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ConnectedRouter
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": undefined,
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <div>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/dashboard"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/demo/:id"
      />
    </div>
  </ConnectedRouter>
</Provider>
`;
